twitter-client.js

```diff
@@ -101,19 +99,29 @@ Twitter.prototype.postFollow = function (params, error, success) {
 
 Twitter.prototype.doRequest = function (url, error, success) {
   url = formatUrl(url);
-
-  this.oauth.get(url, this.accessToken, this.accessTokenSecret, (err, bod, res) => {
-    if(!err && res.statusCode == 200) {
-      limits = {
-        "x-rate-limit-limit": res.headers['x-rate-limit-limit'],
-        "x-rate-limit-remaining": res.headers['x-rate-limit-remaining'],
-        "x-rate-limit-reset": res.headers['x-rate-limit-reset']
-      };
-      success(JSON.parse(bod));
-    } else {
-      console.error('do request error' + err);
+  var arrData = [];
+  var atoken = this.accessToken;
+  var atokensec = this.accessTokenSecret;
+  return new Promise((resolve, reject) => {
+    function cb(err, bod, res) {
+      if(!err) {
+        console.log('success');
+        var jsonBod = JSON.parse(bod);
+        var nextCursor = jsonBod['next_cursor'];
```

Comment:
we want this entire function to have access to `nextCursor`, so let's declare before the `if(!err)`